presto-common/src/main/java/com/facebook/presto/common/type/IntegerEnumType.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.type;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.function.SqlFunctionProperties;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class IntegerEnumType
```

Comment:
Should we call it `LongEnumType` so there's no confusion? Integer can also mean specifically the SQL INTEGER type which is 32-bit (though using 32-bit is probably enough for an enum type so that would be an option as well?)

Reply:
let's go for LongEnumType!